test/python/execution_engine.py

```diff
@@ -0,0 +1,161 @@
+# RUN: %PYTHON %s 2>&1 | FileCheck %s
+
+import gc, sys, os, tempfile
+from aie.ir import *
+from aie.passmanager import *
+from aie.execution_engine import *
+from aie.runtime import *
+
+
+# Log everything to stderr and flush so that we have a unified stream to match
+# errors/info emitted by MLIR to stderr.
+def log(*args):
+    print(*args, file=sys.stderr)
+    sys.stderr.flush()
+
+
+def run(f):
+    log("\nTEST:", f.__name__)
+    f()
+    gc.collect()
+    assert Context._get_live_count() == 0
+
+
+def lowerToLLVM(module):
+    pm = PassManager.parse(
+        "builtin.module(convert-complex-to-llvm,finalize-memref-to-llvm,convert-func-to-llvm,reconcile-unrealized-casts)"
+    )
+    pm.run(module.operation)
+    return module
+
+
+#  Test loading of shared libraries.
+# CHECK-LABEL: TEST: testSharedLibLoad
+def testSharedLibLoad():
+    with Context():
+        module = Module.parse(
+            """
+      module  {
+      func.func @main(%arg0: memref<1xf32>) attributes { llvm.emit_c_interface } {
+        %c0 = arith.constant 0 : index
+        %cst42 = arith.constant 42.0 : f32
+        memref.store %cst42, %arg0[%c0] : memref<1xf32>
+        %u_memref = memref.cast %arg0 : memref<1xf32> to memref<*xf32>
+        call @printMemrefF32(%u_memref) : (memref<*xf32>) -> ()
+        return
+      }
+      func.func private @printMemrefF32(memref<*xf32>) attributes { llvm.emit_c_interface }
+     } """
+        )
+        arg0 = np.array([0.0]).astype(np.float32)
+
+        arg0_memref_ptr = ctypes.pointer(
+            ctypes.pointer(get_ranked_memref_descriptor(arg0))
+        )
+
+        if sys.platform == "win32":
+            shared_libs = [
+                "../../bin/mlir_runner_utils.dll",
+                "../../bin/mlir_c_runner_utils.dll",
+            ]
+        elif sys.platform == "darwin":
+            shared_libs = [
+                "../../lib/libmlir_runner_utils.dylib",
+                "../../lib/libmlir_c_runner_utils.dylib",
+            ]
+        else:
+            shared_libs = [
+                "../../lib/libmlir_runner_utils.so",
+                "../../lib/libmlir_c_runner_utils.so",
+            ]
```

Comment:
I guess this makes some assumptions about directory layout that I don't follow since I get a failure here. Is it possible to get cmake or something else to fill in the path?

Reply:
that directory is `CMAKE_BIN/lib` but but you need to build LLVM/MLIR with the execution engine enabled for them to be copied to that location https://github.com/Xilinx/mlir-aie/blob/8ed3f7b22dfd26f8976589ae215a326b671b0c44/python/CMakeLists.txt#L236

Reply:
Also it's possible if you built without `DCMAKE_PLATFORM_NO_VERSIONED_SONAME=ON` then the `.so` got version and the script doesn't pick it up (fixed in https://github.com/Xilinx/mlir-aie/pull/771).